graphql_compiler/query_pagination/pagination_planning.py

```diff
@@ -1,53 +1,156 @@
 # Copyright 2019-present Kensho Technologies, LLC.
-from collections import namedtuple
+from typing import List, NamedTuple, Tuple
+
+from graphql import DocumentNode
 
 from ..ast_manipulation import get_only_query_definition, get_only_selection_from_ast
-from ..cost_estimation.helpers import is_int_field_type, is_uuid4_type
+from ..cost_estimation.helpers import is_uuid4_type
+from ..cost_estimation.int_value_conversion import field_supports_range_reasoning
 from ..exceptions import GraphQLError
+from ..schema.schema_info import QueryPlanningSchemaInfo
+
+
+class PaginationWarning(object):
```

Comment:
Consider dataclasses for all of these, it would make them easier to serialize and send over the wire.

Also, consider using `advisory` instead of `warning`. Warning is well-defined in Python to be [something quite different](https://docs.python.org/3/library/warnings.html) (e.g. the `Warning` base class is a subclass of `Exception`), so it would be good to avoid confusion. I like `advisory` instead because we are giving the user information and advising (but not requiring) that a certain action be taken, which is the exact meaning of the word per [Merriam Webster](https://www.merriam-webster.com/dictionary/advisory).

Reply:
Good suggestion.